app/src/main/java/com/example/newsapp/ui/home/fragments/EditProfileFragment.kt

```diff
@@ -123,15 +148,36 @@ class EditProfileFragment : Fragment() {
         NavHostFragment.findNavController(this).navigate(action)
     }
 
-    private fun checkPermissionForReadExternalStorage(): Boolean {
-        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
-            val result = context?.checkSelfPermission(Manifest.permission.READ_EXTERNAL_STORAGE)
-            return result == PackageManager.PERMISSION_GRANTED
+    private fun setupPermissions(): Boolean {
+        val permission = ContextCompat.checkSelfPermission(requireActivity(),
+            Manifest.permission.CAMERA)
+
+        if (permission != PackageManager.PERMISSION_GRANTED) {
+            UtilityFunctions.printLogs(getString(R.string.error),
+                getString(R.string.permission_error))
+            return false
         }
-        return false
+        return true
+    }
+
+    private fun makeRequest() {
+        ActivityCompat.requestPermissions(requireActivity(),
+            arrayOf(Manifest.permission.CAMERA),
+            Constants.CAMERA_REQUEST_CODE)
     }
 
 
+    private fun getBitmapFromUri(filePath: Uri) {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
+            requireContext().contentResolver?.let {
+                val source =
+                    ImageDecoder.createSource(requireContext().contentResolver, filePath)
```

Comment:
@jnandrawos you can use "it" here directly

Reply:
done